pkg/providers/cloudstack/validator.go

```diff
@@ -135,6 +136,9 @@ func (v *Validator) ValidateClusterMachineConfigs(ctx context.Context, clusterSp
 		if err := v.validateMachineConfig(ctx, clusterSpec.CloudStackDatacenter, machineConfig); err != nil {
 			return fmt.Errorf("machine config %s validation failed: %v", machineConfig.Name, err)
 		}
+		if err := v.validateTemplateMatchesKubernetesVersion(ctx, machineConfig, clusterSpec); err != nil {
```

Comment:
I had to split validateTemplateMatchesKubernetesVersion method separately to avoid cyclomatic complexity error.